### PR TITLE
feat: Switch to proper RAG for @ecl chat agent

### PR DESCRIPTION
**Pull request by GordonSmith**

### CODE REVIEW

File: util/gather-docs.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+function convertWindowsToLinuxPath(windowsPath) {
+    // Normalize the path to handle mixed slashes
+    const normalizedPath = path.normalize(windowsPath);
+    // Convert to POSIX format
+    return path.posix.join(...normalizedPath.split(path.sep)).split("G:/").join("/mnt/g/");
+}
+
+async function convertDocBookToMarkdown(inputFile: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+        const pandoc = spawn("wsl", ["pandoc", "-f", "docbook", "-t", "markdown", "-s", convertWindowsToLinuxPath(inputFile)]);
```
Comment (jeclrsg) [inline util/gather-docs.ts:14]: Where does pandoc come from?

File: util/gather-docs.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import { readdir, stat } from "fs/promises";
+import * as path from "path";
+import { spawn } from "child_process";
+
+function convertWindowsToLinuxPath(windowsPath) {
+    // Normalize the path to handle mixed slashes
+    const normalizedPath = path.normalize(windowsPath);
+    // Convert to POSIX format
+    return path.posix.join(...normalizedPath.split(path.sep)).split("G:/").join("/mnt/g/");
```
Comment (jeclrsg) [inline util/gather-docs.ts:9]: I couldn't find where this path for a G drive was defined?

File: util/index-docs.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import { MemoryVectorStore } from "langchain/vectorstores/memory";
+import { RecursiveCharacterTextSplitter } from "@langchain/textsplitters";
+import { OllamaEmbeddings } from "@langchain/ollama";
+import { Zstd } from "@hpcc-js/wasm-zstd";
+import { findXmlFiles as walkDocs } from "./gather-docs";
+
+const modelPath = path.join(".", "docs", "models", "bge-base-en-v1.5-q4_k_m.gguf");
+
+function serializeStore(store: MemoryVectorStore): string {
+    const serializData = {
```
Comment (jeclrsg) [inline util/index-docs.ts:12]: just a typo, serializeData

**Review by jeclrsg — COMMENTED** [review]
@GordonSmith had a couple questions about how that pandoc spawned process was working

File: util/gather-docs.ts (hunk 1 above)
Comment (GordonSmith) [reply util/gather-docs.ts:14]: `apt install pandoc`, Ideally the docs.vecdb file would be generated as part of CI / Release building, but it really needs a GPU...

Comment (GordonSmith) [reply util/gather-docs.ts:14]: Also it should really be done in the HPCC-Platform repo.
FWIW the code in the "/utils" folder is only for periodic execution and the output of those runs get committed to GitHub - similar to how we do the wsdl stuff in the Viz comms layer.

File: util/gather-docs.ts (hunk 2 above)
Comment (GordonSmith) [reply util/gather-docs.ts:9]: Yea very specific to my machine (for now).

Comment (GordonSmith) [reply util/gather-docs.ts:9]: FWIW - I keep my "dev" folders in a ReFS virtual disk mounted as "G" on all my machines:
image

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**